### PR TITLE
Feat: create user with wallet in single request

### PR DESCRIPTION
**Pull request by caudurodev**
- handles creation of user with wallet with one signing event (new endpoint from backend https://docs.dfns.co/d/api-docs/authentication/registration/completeenduserregistration

### CODE REVIEW

File: sdk-connect/DfnsConnectProvider.tsx [hunk 1]
```diff
@@ -151,6 +152,26 @@ export const DfnsConnectProvider: React.FC<PropsWithChildren> = ({
...
+      return await _sendMessageToIframe({
+        action: MessageActions.createUserAndWallet,
+        actionResponse: MessageActionsResponses.createUserAndWalletSuccess,
+        challenge,
+        walletName,
+        network,
+        showScreen,
+      });
+    } catch (e) {
+      console.error("error logging out");
```
Comment (adriendfns) [inline sdk-connect/DfnsConnectProvider.tsx:171]: are we really logging out?

File: sdk-connect/index.ts [hunk 2]
```diff
@@ -138,6 +121,59 @@ export type SignRegisterUserInitProps = {
   challenge: UserRegistrationChallenge;
   showScreen?: IframeActiveState;
 };
+
+export type CreateUserAndWalletProps = {
+  network:
+    | "Algorand"
```
Comment (adriendfns) [inline sdk-connect/index.ts:127]: any `BlockchainNetwork` type available?

File: sdk-connect/windowMessage.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ export interface CreateWalletPayload {
   userName?: string;
   walletName?: string;
   networkId?: BlockchainNetwork;
+  network?: string;
```
Comment (adriendfns) [inline sdk-connect/windowMessage.ts:48]: `networkId` and `network` seems redundant?

File: sdk-connect/index.ts (hunk 2 above)
Comment (caudurodev) [reply sdk-connect/index.ts:127]: I don't believe so - Asked Dan about it and this was the solution for now.

File: sdk-connect/DfnsConnectProvider.tsx (hunk 1 above)
Comment (caudurodev) [reply sdk-connect/DfnsConnectProvider.tsx]: No. Forgot to update some of these. Updated a few more thanks. updated.

File: sdk-connect/index.ts (hunk 2 above)
Comment (caudurodev) [reply sdk-connect/index.ts:127]: Movig this list to a type for now.

**Review by ncornwell — APPROVED** [review]

**Pull request merged**